src/test/java/com/answerdigital/benhession/academy/answerkingweek2/controllers/CategoryControllerTest.java

```diff
@@ -0,0 +1,113 @@
+@ExtendWith(SpringExtension.class)
+class CategoryControllerTest {
+    @Autowired
+    private CategoryController categoryController;
+
+    @MockBean
+    private CategoryService categoryService;
+
+    /**
+     * Method under test: {@link CategoryController#addCategory(AddCategoryRequest)}
+     */
+    @Test
+    void testAddCategory() throws Exception {
+        MockHttpServletRequestBuilder contentTypeResult = MockMvcRequestBuilders.get("/")
+                .contentType(MediaType.APPLICATION_JSON);
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        MockHttpServletRequestBuilder requestBuilder = contentTypeResult.content(
+                objectMapper.writeValueAsString(new AddCategoryRequest("Name", "The characteristics of someone or something")));
+        ResultActions actualPerformResult = MockMvcBuilders.standaloneSetup(categoryController)
+                .build()
+                .perform(requestBuilder);
+        actualPerformResult.andExpect(MockMvcResultMatchers.status().isNotFound());
+    }
+
+    /**
+     * Method under test: {@link CategoryController#addItemToCategory(Long, Long)}
+     */
+    @Test
+    void testAddItemToCategory() throws Exception {
+        MockHttpServletRequestBuilder requestBuilder = MockMvcRequestBuilders.put("/{categoryId}/additem/{itemId}", 123L,
+                123L);
+        ResultActions actualPerformResult = MockMvcBuilders.standaloneSetup(categoryController)
+                .build()
+                .perform(requestBuilder);
+        actualPerformResult.andExpect(MockMvcResultMatchers.status().isNotFound());
+    }
+
+    /**
+     * Method under test: {@link CategoryController#deleteCategory(Long)}
+     */
+    @Test
+    void testDeleteCategory() throws Exception {
+        MockHttpServletRequestBuilder requestBuilder = MockMvcRequestBuilders.delete("/{categoryId}", 123L);
+        ResultActions actualPerformResult = MockMvcBuilders.standaloneSetup(categoryController)
+                .build()
+                .perform(requestBuilder);
+        actualPerformResult.andExpect(MockMvcResultMatchers.status().isNotFound());
+    }
+
+    /**
+     * Method under test: {@link CategoryController#getAllCategories()}
+     */
+    @Test
+    void testGetAllCategories() throws Exception {
+        MockHttpServletRequestBuilder requestBuilder = MockMvcRequestBuilders.get("/");
+        ResultActions actualPerformResult = MockMvcBuilders.standaloneSetup(categoryController)
+                .build()
+                .perform(requestBuilder);
+        actualPerformResult.andExpect(MockMvcResultMatchers.status().isNotFound());
+    }
+
+    /**
+     * Method under test: {@link CategoryController#removeItemFromCategory(Long, Long)}
+     */
+    @Test
+    void testRemoveItemFromCategory() throws Exception {
+        MockHttpServletRequestBuilder requestBuilder = MockMvcRequestBuilders.put("/{categoryId}/removeitem/{itemId}",
```

Comment:
I think this two-liner would look much simpler to understand: 
@Test
    @WithMockUser("paul")
    void testRemoveItemFromCategory() throws Exception {
        mockMvc.perform(put("/{categoryId}/removeitem/{itemId}",
                123L, 123L)).andExpect(MockMvcResultMatchers.status().isNotFound());
    }

Similarly for all other tests as well.